libr/bin/format/mach0/mach0.c

```diff
@@ -2949,32 +2946,42 @@ const struct symbol_t *MACH0_(get_symbols)(struct MACH0_(obj_t) *bin) {
 		for (i = 0; i < bin->nsymtab; i++) {
 			struct MACH0_(nlist) *st = &bin->symtab[i];
 			stridx = st->n_strx;
+			ut64 addr = st->n_value;
 			// 0 is for imports
 			// 1 is for symbols
 			// 2 is for func.eh (exception handlers?)
 			int section = st->n_sect;
-			if (section == 1 && j < symbols_count) { // text ??st->n_type == 1)
-			//if (j < symbols_count) { // text ??st->n_type == 1)
+			if (section == 1 && j < symbols_count) {
+				// check if symbol exists already
 				/* is symbol */
 				symbols[j].addr = st->n_value;
 				symbols[j].offset = addr_to_offset (bin, symbols[j].addr);
 				symbols[j].size = 0; /* find next symbol and crop */
-				if (st->n_type & N_EXT) {
-					symbols[j].type = R_BIN_MACH0_SYMBOL_TYPE_EXT;
-				} else {
-					symbols[j].type = R_BIN_MACH0_SYMBOL_TYPE_LOCAL;
-				}
+				symbols[j].type = (st->n_type & N_EXT)
+					? R_BIN_MACH0_SYMBOL_TYPE_EXT
+					: R_BIN_MACH0_SYMBOL_TYPE_LOCAL;
 				char *sym_name = get_name (bin, stridx, false);
 				if (sym_name) {
 					symbols[j].name = sym_name;
 				} else {
-					symbols[j].name = r_str_newf ("entry%d\n", i);
-					//symbols[j].name[0] = 0;
+					size_t k = 0;
+					bool named_symbol_found = false;
+					symbols[j].last = true;
+					for (k = 0; !symbols[k].last; k++) {
```

Comment:
depending on the binary this could slow down a lot the parsing of mach0 files, but probably empty `sym_name` is rare enough to not care about this case for now.

Reply:
Yes, this is not a good solution. Actually i couldnt find a bin with unnamed symbols so not sure if we could just get rid of the whole loop entirely without losing any valuable information. Maybe stripped?